verifier/graph.go

```diff
@@ -254,19 +254,23 @@ func (g *Graph) IsRoot(c *x509.Certificate) bool {
 }
 
 // AppendFromPEM adds any certificates encoded as PEM from r to the graph. If
-// root is true, it marks them as roots. It returns the number of certificates
-// parsed.
-func (g *Graph) AppendFromPEM(r io.Reader, root bool) int {
+// root is true, it marks them as roots. It returns:
+// - The number of certificates parsed
+// - A slice of errors encountered while parsing certificates
+// - Any non-EOF error encountered reading from r
+func (g *Graph) AppendFromPEM(r io.Reader, root bool) (int, []error, error) {
```

Comment:
To maintain API backwards compatibility, how about renaming this func as you have it written, to `AppendFromPEMReportErr` (or something like that), and then have `AppendFromPEM` wrap that function, but ignore the error returns (i.e., keep the original function signature).  That way existing consumers of the API outside this project won't it an API change.